src/AutoMerge/pull-requests.jl

```diff
@@ -95,51 +99,40 @@ function pull_request_build(api::GitHub.GitHubAPI,
-                           registry;
-                           auth = auth,
-                           authorized_authors=authorized_authors,
-                           authorized_authors_special_jll_exceptions=authorized_authors_special_jll_exceptions,
-                           error_exit_if_automerge_not_applicable = error_exit_if_automerge_not_applicable,
-                           registry_head = registry_head,
-                           registry_master = registry_master,
-                           suggest_onepointzero = suggest_onepointzero,
-                           whoami=whoami,
-                           registry_deps = registry_deps)
-        rm(registry_master; force = true, recursive = true)
+        registration_type = NewPackage()
     elseif is_new_version(pr)
-        registry_master = clone_repo(registry)
-        if !master_branch_is_default_branch
-            checkout_branch(registry_master, master_branch)
-        end
-        pull_request_build(api,
-                           NewVersion(),
-                           pr,
-                           current_pr_head_commit_sha,
-                           registry;
-                           auth = auth,
-                           authorized_authors=authorized_authors,
-                           authorized_authors_special_jll_exceptions=authorized_authors_special_jll_exceptions,
-                           error_exit_if_automerge_not_applicable = error_exit_if_automerge_not_applicable,
-                           registry_head = registry_head,
-                           registry_master = registry_master,
-                           suggest_onepointzero = suggest_onepointzero,
-                           whoami=whoami,
-                           registry_deps = registry_deps)
-        rm(registry_master; force = true, recursive = true)
+        registration_type = NewVersion()
     else
         throw_not_automerge_applicable(
             AutoMergeNeitherNewPackageNorNewVersion,
             "Neither a new package nor a new version. Exiting...";
             error_exit_if_automerge_not_applicable = error_exit_if_automerge_not_applicable,
         )
+        return nothing
+    end
+
+    registry_master = clone_repo(registry)
+    if !master_branch_is_default_branch
+        checkout_branch(registry_master, master_branch)
     end
+    pkg, version = parse_pull_request_title(registration_type, pr)
+    data = GitHubAutoMergeData(;api = api,
+                               registration_type = registration_type,
+                               pr = pr,
+                               pkg = pkg,
+                               version = version,
+                               current_pr_head_commit_sha = current_pr_head_commit_sha,
+                               registry = registry,
+                               auth = auth,
+                               authorized_authors_authors = authorized_authors_authors,
```

Comment:
```suggestion
                               authorized_authors = authorized_authors,
```
found by the integration tests

Reply:
That's a weird mistake. (Well, the right hand side was done by a keyboard macro, so the original typo was only made once.)